facebook/ThreatExchange
# Index multi reader

Summary
---------

Make the PDQ indexer read from all `.pdq.te` files and merge them before building the PDQ index

Test Plan
---------

To test AWS reading writing works correctly:
1. `terraform apply`
2. Push new docker image to aws (refresh `fetcher` and `pdq_indexer` functions)
3. Delete the `threat_exchange_data/` and `index/` folders 
4. Watch them repopulate after fetcher runs

To test merging works correctly
1. Make sure `threat_exchange_data/` has files for the two test configs. One file should be a superset of the other
2. locally run the pdq_indexer (`python3 -m hmalib.lambdas.pdq.pdq_indexer`) and look (using `print`) to see that the merged index has the right number of hashes and that the privacy_groups were correctly merged
